Doc/c-api/weakref.rst

```diff
@@ -51,10 +51,21 @@ as much as it can.
    ``None``, or ``NULL``, this will return ``NULL`` and raise :exc:`TypeError`.
 
 
+.. c:function:: int PyWeakref_GetRef(PyObject *ref, PyObject **pobj)
+
+   Get a :term:`strong reference` to the referenced object from a weak
+   reference, *ref*, into *\*pobj*.
+   Return 0 on success. Raise an exception and return -1 on error.
```

Comment:
Let's consider returning 1 if `**pobj` was set to an object, and 0 if it's set to NULL: distinguishing “missing result” from “full sucess” in the return value miiiiight be better API in general (especially for a `PyIter_NextItem`), and maaaaaybe it'd be worth the extra check to be consistent.
I [posted about this elsewhere](https://github.com/python/cpython/issues/105201#issuecomment-1600491761), let's not block the PR on it. There's enough time to change it in 3.13.

Reply:
Since the devguide got updated to include this new guideline, I created https://github.com/python/cpython/pull/106561 to return 1 on success.